File: test77_out/test77/Test_21/Test_21.c

/*
 * ======== string literals ========
 */

static const char test77_Test_21__s__fileName[] = "Test_21.p";
static const char test77_Test_21__s__5[] = "Test_21";
static const char test77_Test_21__s__3[] = "Running!...\n";
static const char test77_Test_21__s__2[] = "TICK3!\n";
static const char test77_Test_21__s__4[] = "Shutdown!\n";
static const char test77_Test_21__s__0[] = "TICK1!\n";
static const char test77_Test_21__s__1[] = "TICK2!\n";

/*
 * ======== code injections ========
 */

 #include <stdlib.h> 

/*
 * ======== function definitions ========
 */

void test77_Test_21_tick3__I() {
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__2);
}

void test77_Test_21_pollen__run__E() {
    pollen_time_Timer_Timer_start__E(&(test77_Test_21_t1__V), 250, true);
    pollen_time_Timer_Timer_start__E(&(test77_Test_21_t2__V), 350, true);
    pollen_time_Timer_Timer_start__E(&(test77_Test_21_t3__V), 550, true);
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__3);
    pollen_event_Newsroom_run__E();
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__3);
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__3);
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__3);
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__3);
    test77_Test_21_pollen__shutdown__E(0);
}

void test77_Test_21_tick2__I() {
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__1);
}

void test77_Test_21_tick1__I() {
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__0);
}

void test77_Test_21_pollen__shutdown__E( uint8 id ) {
    test77_Test_21_pollen__printStr((string)test77_Test_21__s__4);
     exit(0) ;

}

void test77_Test_21_targetInit__I() {
}

File: test77_out/test77/Test_21/Test_21.js

/*
 * ======== MODULE test77.Test_21 ========
 */

var $$c = function() {
    const Test_21 = {};
    Test_21.Test_21 = Test_21
    Test_21.$name = 'test77.Test_21';
    Test_21.pollen$used = 0;
    Test_21.Timer = $units['pollen.time.Timer'];
    Test_21.tick1 = new $$Ref('test77_Test_21_tick1__I');
    Test_21.tick2 = new $$Ref('test77_Test_21_tick2__I');
    Test_21.tick3 = new $$Ref('test77_Test_21_tick3__I');
    Test_21.pollen__run = new $$Ref('test77_Test_21_pollen__run__E');
    Test_21.pollen__shutdown = new $$Ref('test77_Test_21_pollen__shutdown__E');
    Test_21.targetInit = new $$Ref('test77_Test_21_targetInit__I');
    Test_21.$$hostInit = function() {
        var $$text = '';
        Test_21.t1 = new $units['pollen.time.Timer'].Timer('test77_Test_21_t1__V'); Test_21.t1.new_host((Test_21.tick1));
        Test_21.t2 = new $units['pollen.time.Timer'].Timer('test77_Test_21_t2__V'); Test_21.t2.new_host((Test_21.tick2));
        Test_21.t3 = new $units['pollen.time.Timer'].Timer('test77_Test_21_t3__V'); Test_21.t3.new_host((Test_21.tick3));
        return $$text;
    }
    Test_21.pollen__uses__ = function() {
        $$bind($units['virtual.mcu.GlobalInterrupts'], 'pollen$used', true);
        $$bind($units['virtual.mcu.TimerMilli'], 'pollen$used', true);
        $$bind($units['pollen.event.Newsroom'], 'pollen$used', true);
        $$bind($units['pollen.time.Timer'], 'pollen$used', true);
    }
    Test_21.$$privateInit = function() {
    }
    return Test_21;
}
var $$u = $$c();
$units['test77.Test_21'] = $$u;
$units.push($$u);

